core/src/executor/execute.rs

```diff
@@ -89,9 +93,11 @@ pub async fn execute<T: GStore + GStoreMut>(
 }
 
 async fn execute_inner<T: GStore + GStoreMut>(
-    storage: &mut T,
+    storage: Option<&mut T>,
     statement: &Statement,
 ) -> Result<Payload> {
+    let storage = storage.ok_or(ExecuteError::StatelessOperation)?;
```

Comment:
The purpose of making the execution layer functions to take `storage` param as optional is for handling `None` case.
Then, we can execute `SELECT 1;` like queries which does not require storage by not using storage.
However, in this case, because
```rust
let storage = storage.ok_or(ExecuteError::StatelessOperation)?;
```
the above line requires `storage` not to be optional. we can't still handle `SELECT 1` without storage.

if the entire function content requires `storage` then we don't need to make the function `storage` param optional.
once we make the storage optional, then it means that we also have some cases which `None` works for that.